clash-prelude/src/Clash/Sized/Internal/BitVector.hs

```diff
@@ -847,21 +847,30 @@ reduceAnd# bv@(BV 0 i) = Bit 0 (W# (int2Word# (dataToTag# check)))
 
     sz    = natVal bv
     maxI  = (2 ^ sz) - 1
-reduceAnd# bv = V.foldl (.&.) 1 (V.bv2v bv)
+reduceAnd# bv@(BV m i) = Bit (W# (int2Word# (dataToTag# check))) 0
+  where
```

Comment:
Comment for the case where `m == 1`: I had to work out an example to convince myself this trick was correct. Could you add a comment?

Reply:
> the case where `m == 0`

Oh man, that (thoroughly hashed) code is ancient. It took quite some archeology. A young fresh-faced Christiaan apparently wrote it in 2014: b8f2cc13f. I presume to attain _ludricous speed mode_. There's definitely [unneeded code](https://github.com/clash-lang/clash-compiler/blob/2e8d98f11ade4235fc5d437e3752cbd2877798f8/clash-prelude/src/Clash/Class/BitPack/Internal.hs#L154-L157) in `boolToBit` for this simple, non-XExcepty case.

> the case where `m == 1`

(`m /= 0`, you're probably thinking of `Bit`, this is `BitVector`)

Hah, I wondered whether the "trickery" passed the threshold of requiring a comment or not! I figured since we do more bit operations that really make you go "wait, how does that work?" initially that are not commented, I'd leave it out and see what happened in review. Sure, I'll add a comment.

Reply:
You're right, I meant `m /= 0`

>  I presume to attain ludricous speed mode.

Good enough for me :-). This thread can be closed IMO, comment looks good.